### PR TITLE
Add mesos in the CI

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

### CODE REVIEW

File: test/integration/helpers.bash [hunk 1]
```diff
@@ -165,7 +165,7 @@ function start_docker() {
 		# We have to manually call `hostname` since --hostname and --net cannot
 		# be used together.
 		DOCKER_CONTAINERS[$i]=$(
-			docker_host run -d --name node-$i --privileged -it --net=host \
+			docker_host run -d --name node-$i --privileged -v /usr/local/bin -v /var/run/docker.sock -it --net=host \
```
Comment (vieux) [inline test/integration/helpers.bash]: we do a `--volume-from` in the mesos tests.

**Comment by chanwit** [comment]
This is super!!!

**Comment by vieux** [comment]
:fireworks: :cool: the CI is green

@tnachen @jimenez

**Comment by jimenez** [comment]
:+1:

**Comment by tnachen** [comment]
Nice! I'm watching the results to see if CI passes

**Comment by vieux** [comment]
@tnachen it will :D it was green a minute ago, I had to do a force push

**Comment by tnachen** [comment]
@vieux Cool, is the mesos-master/mesos-slave built with the latest master (0.23)?

**Comment by vieux** [comment]
@tnachen yes, I used @jimenez's image that was build today with master (0.23)

File: test/integration/mesos/mesos_helpers.bash [hunk 2]
```diff
@@ -2,11 +2,36 @@
 
 load ../helpers
 
-MESOS_CLUSTER_ENTRYPOINT=${MESOS_CLUSTER_ENTRYPOINT:-0.0.0.0:5050}
+MESOS_IMAGE=jimenez/mesos-dev:clang
+MESOS_MASTER_PORT=$(( ( RANDOM % 1000 )  + 7000 ))
```
Comment (aluzzardi) [inline test/integration/mesos/mesos_helpers.bash]: different port please (7000 is already taken)

File: test/integration/mesos/mesos_helpers.bash [hunk 3]
```diff
@@ -2,11 +2,36 @@
...
+
+# Start mesos master and slave.
+function start_mesos() {
+	MESOS_MASTER=$(
+	docker_host run -d --name mesos-master --net=host \
+	$MESOS_IMAGE /mesos/build/bin/mesos-master.sh --ip=127.0.0.1 --work_dir=/ --registry=in_memory --port=$MESOS_MASTER_PORT
+		)
+
+	MESOS_SLAVE=$(
+	docker_host run -d --name mesos-slave --volumes-from node-0 -e DOCKER_HOST="${HOSTS[0]}" -v /sys/fs/cgroup:/sys/fs/cgroup --net=host \
```
Comment (tnachen) [inline test/integration/mesos/mesos_helpers.bash]: You should also run --pid=host, and containerizers=docker

Comment (tnachen) [inline test/integration/mesos/mesos_helpers.bash]: Also one thing I usually do is to mount in the sandbox volume from the host, so like -v /tmp/mesos:/tmp/mesos
Doing this we can also try out restarting the slave and it should recover all the docker tasks with swarm, etc.

File: test/integration/mesos/mesos_helpers.bash [hunk 4]
```diff
@@ -2,11 +2,36 @@
...
+	$MESOS_IMAGE /mesos/build/bin/mesos-slave.sh --master=127.0.0.1:$MESOS_MASTER_PORT --containerizers=docker --hostname=127.0.0.1 --port=$(($MESOS_MASTER_PORT + 1))
```
Comment (tnachen) [inline test/integration/mesos/mesos_helpers.bash]: I just added some new flags that you should definitely use for running mesos-slave in a docker contianer.
You should at least use --docker-mesos-image=$MESOS_IMAGE
This will create all the executors in docker containers, and you can recover all the tasks on slave restart.

File: test/integration/mesos/mesos_helpers.bash (hunk 3 above)
Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: I agree on the sandbox, can you tell me more about the `--pid=host`, why is this needed ?

File: test/integration/mesos/mesos_helpers.bash (hunk 4 above)
Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: I don't think it will work, the mesos-slave can't access the image, here is the current setup:

```
jenkins 
 └ docker (here the image jimenez/mesos-dev:clang is pulled)
   └ docker (there the image is not)
   └ mesos-slave 
   └ mesos-master
```

File: test/integration/mesos/mesos_helpers.bash (hunk 3 above)
Comment (tnachen) [inline test/integration/mesos/mesos_helpers.bash]: Mesos will try to find your cgroup settings based on the pid of the docker container you launched, so without --pid=host DockerContainerizer can't see the Docker container that you launched unless you're also using docker in docker where docker daemon is in the same image.

Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: As `--pid=host` was added recently, I added`--privileged` so it will work with older version.

File: test/integration/mesos/mesos_helpers.bash (hunk 4 above)
Comment (vieux) [inline test/integration/mesos/mesos_helpers.bash]: Let's add those new flags when we need them, it will require some changes in the tests: if a test does
```
docker run busybox ls
docker images
```

I'm expecting to see only one image (`busybox`), not two (`busybox`, `jimenez/mesos-dev:clang`)
